app/scripts/lib/logger.js

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Shared logging module that performs the checks and string
+ * interpolation.
+ *
+ */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  /**
+   * Constructor of log module.
+   *
+   * @param logWindow Window object that contains console used for logging.
+   *
+   * @constructor
+   */
+  function Logger(logWindow) {
+    this._window = logWindow || window;
+  }
+
+  Logger.prototype = {
+
+    /**
+     * Wrapper for `console.info` function that checks for availability and
+     * then prints info.
+     *
+     */
+    info: function () {
+      if (this._window.console && this._window.console.info) {
+        this._window.console.info.apply(this._window.console, arguments);
+      }
+    },
+
+    /**
+     * Wrapper for `console.trace` function that checks for availability and then prints
+     * trace.
+     *
+     */
+    trace: function () {
+      if (this._window.console && this._window.console.trace) {
+        this._window.console.trace();
+      }
+    },
+
+    /**
+     * Wrapper for `console.warn` function that checks for availability and
+     * then prints warn.
+     *
+     */
+    warn: function () {
+      if (this._window.console && this._window.console.warn) {
+        this._window.console.warn.apply(this._window.console, arguments);
+      }
+    },
+
+    /**
+     * Wrapper for `console.error` function that checks for availability and interpolates
+     * error messages if a translation exists.
+     *
+     * @param {error} Error object with optional errorModule.
+     */
+    error: function (error) {
+      if (this._window.console && this._window.console.error) {
+        var errorMessage = '';
+
+        // If error module is present, attempt to interpolate string, else use error object message
+        if (error.errorModule) {
+          errorMessage = error.errorModule ? error.errorModule.toInterpolatedMessage(error) : String(error);
```

Comment:
This ternary looks like it is repeating the same check that the `if` does. Is it redundant or am I missing something?